src/plugins/intel_npu/src/backend/src/zero_pipeline.cpp

```diff
@@ -149,92 +149,109 @@ struct IntegratedPipeline final : public Pipeline {
                        std::shared_ptr<zeroProfiling::NpuInferProfiling> npu_profiling,
                        CommandQueue& command_queue,
                        const uint32_t& group_ordinal,
-                       std::unordered_map<std::string, std::shared_ptr<ov::ITensor>>& tensors)
+                       std::unordered_map<std::string, std::shared_ptr<ov::ITensor>>& tensors,
+                       const size_t batch_size)
         : _config(config),
           _command_queue{command_queue},
-          _command_list{device_handle, context, graph_ddi_table_ext, _config, group_ordinal},
-          _fence{_command_queue, _config},
-          _event_pool{device_handle, context, 1, _config},
-          _event{_event_pool.handle(), 0, _config},
+          _event_pool{device_handle, context, batch_size ? static_cast<uint32_t>(batch_size) : 1, _config},
```

Comment:
Not sure about this one. In case we plan to change implementation and using 0 as default for batch size or something similar we still need an event pool for running. So in that particular case, we would still need an event pool here.

If there is an error, the user should be notified earlier.

Reply:
Can we use `batch_size == 1` as the default?

Reply:
we can do that, but still need to check batch_size to not be set to 0 somewhere else.

Reply:
Yes, the check is required.

I just want to emphasize that batch_size == 0 looks like an error situation.